django/db/models/fields/mixins.py

```diff
@@ -1,31 +1,52 @@
+import warnings
+
 from django.core import checks
+from django.utils.deprecation import RemovedInDjango60Warning
+from django.utils.functional import cached_property
 
 NOT_PROVIDED = object()
 
 
 class FieldCacheMixin:
-    """Provide an API for working with the model's fields value cache."""
+    """
+    Provide an API for working with the model's fields value cache.
```

Comment:
Since we are already modifying this docstring, I would advice we follow PEP and define a class as a noun and not as a verb:
```suggestion
    An API for working with the model's fields value cache.
```

Reply:
Thanks, done.